Modix.Services/UserInfo/UserInfoCommand.cs

```diff
@@ -0,0 +1,297 @@
+﻿#nullable enable
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Discord;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Modix.Data;
+using Modix.Data.Extensions;
+using Modix.Data.Models.Core;
+using Modix.Data.Models.Moderation;
+using Modix.Services.AutoRemoveMessage;
+using Modix.Services.Extensions;
+using Modix.Services.Images;
+using Modix.Services.Utilities;
+
+namespace Modix.Services.UserInfo
+{
+    public class UserInfoCommand : IRequest
+    {
+        public UserInfoCommand(IMessage message, ulong guildId, ulong userId)
```

Comment:
I'm not saying I'm against using Mediator here, but I wanna put forth a quick point:

Personally, I don't see any point in factoring a simple method call into a `Request`/`IRequestHandler` pair, if we're not going to use any of the other features of request pipelines. Seems like a layer of abstraction that accomplishes nothing, unless we're actually gonna attach pre-request handlers or whatnot.

I'm not familiar with what's really possible with the pipelines API, but if it's possible, I think it would be good to actually put the primary request handler in `Modix.Services`, and decouple it entirely from rendering logic in `Modix.Bot` which could (maybe) be implemented as a post-request handler. With this pattern, we could more easily implement "commands" where the business logic is shared by both the Bot and Web frontends. The way we do this now is just by having methods on services that represent "user-facing operations" sitting right alongside methods that are "inter-service operations" and its quite messy.

Reply:
Yeah I'm not sure this is a perfect application for the command - the idea would be the web controller could easily call out to here, but the handler itself is already intrinsically hooked to the Discord client/bot and wouldn't be generic enough for the web, though it could be changed.

Reply:
I wouldn't say commands should be decoupled from Discord, just decoupled from the "bot frontend". For a command like this, that REALLY only means rendering.